### ISSUE TITLE
Note SDOs do not work on custom STIX types

### ISSUE THREAD

**Issue by isaisabel** [oasis-open/cti-python-stix2#501]
As a user of cti-python-stix2, I want to be able to work with custom STIX types with the same level of functionality as I can with stock STIX types.

Currently, the `stix2` library throws an exception when loading STIX bundles containing a note referencing a custom STIX type. See the example below.

### Example STIX Bundle:

The following bundle contains a single [x-mitre-tactic](https://github.com/mitre/cti/blob/master/USAGE.md#tactics) object and an attached note.

```json
{
    "type": "bundle",
    "id": "bundle--ad5f3bce-004b-417e-899d-392f8591ab55",
    "spec_version": "2.1",
    "objects": [
        {
            "id": "x-mitre-tactic--d108ce10-2419-4cf9-a774-46161d6c6cfe",
            "created_by_ref": "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5",
            "name": "Collection",
            "description": "The adversary is trying to gather data of interest to their goal.\n\nCollection consists of techniques adversaries may use to gather information and the sources information is collected from that are relevant to following through on the adversary's objectives. Frequently, the next goal after collecting data is to steal (exfiltrate) the data. Common target sources include various drive types, browsers, audio, video, and email. Common collection methods include capturing screenshots and keyboard input.",
            "external_references": [
                {
                    "external_id": "TA0009",
                    "url": "https://attack.mitre.org/tactics/TA0009",
                    "source_name": "mitre-attack"
                }
            ],
            "object_marking_refs": [
                "marking-definition--fa42a846-8d90-4e51-bc29-71d5b4802168"
            ],
            "x_mitre_shortname": "collection",
            "type": "x-mitre-tactic",
            "modified": "2019-07-19T17:44:53.176Z",
            "created": "2018-10-17T00:14:20.652Z",
            "spec_version": "2.1",
            "x_mitre_domains": [
                "enterprise-attack"
            ],
            "x_mitre_version": "1.0"
        },
        {
            "type": "note",
            "id": "note--559fb05e-f4fc-4f93-9caa-e76f6aeb0207",
            "spec_version": "2.1",
            "abstract": "Tactic note example",
            "content": "Velit non dolore dolor. Quaerat quisquam dolor sit ipsum modi modi. Magnam porro voluptatem etincidunt adipisci sit tempora ut. Amet consectetur ipsum sit eius etincidunt. Dolorem porro consectetur tempora quaerat aliquam tempora dolore. Ut sit neque eius. Consectetur quisquam etincidunt dolor tempora dolore.",
            "object_refs": [
                "x-mitre-tactic--d108ce10-2419-4cf9-a774-46161d6c6cfe"
            ]
        }
    ]
}
```
### Example error on loading that bundle:
```
$ python3
>>> from stix2 import MemoryStore
>>> src = MemoryStore()
>>> src.load_from_file("tactic-note-bundle.json")
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/stix2/base.py", line 104, in _check_property
    kwargs[prop_name] = prop.clean(kwargs[prop_name])
  File "/usr/local/lib/python3.8/site-packages/stix2/properties.py", line 234, in clean
    result = [
  File "/usr/local/lib/python3.8/site-packages/stix2/properties.py", line 235, in <listcomp>
    self.contained.clean(item)
  File "/usr/local/lib/python3.8/site-packages/stix2/properties.py", line 509, in clean
    raise ValueError("The type-specifying prefix '%s' for this property is not valid" % (possible_prefix))
ValueError: The type-specifying prefix 'x-mitre-tactic' for this property is not valid

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/stix2/datastore/memory.py", line 146, in load_from_file
    return self.source.load_from_file(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/stix2/datastore/memory.py", line 362, in load_from_file
    _add(self, stix_data, self.allow_custom, version)
  File "/usr/local/lib/python3.8/site-packages/stix2/datastore/memory.py", line 40, in _add
    _add(store, stix_obj, allow_custom, version)
  File "/usr/local/lib/python3.8/site-packages/stix2/datastore/memory.py", line 47, in _add
    stix_obj = parse(stix_data, allow_custom, version)
  File "/usr/local/lib/python3.8/site-packages/stix2/parsing.py", line 47, in parse
    obj = dict_to_stix2(obj, allow_custom, version)
  File "/usr/local/lib/python3.8/site-packages/stix2/parsing.py", line 142, in dict_to_stix2
    return obj_class(allow_custom=allow_custom, **stix_dict)
  File "/usr/local/lib/python3.8/site-packages/stix2/base.py", line 198, in __init__
    self._check_property(prop_name, prop_metadata, setting_kwargs)
  File "/usr/local/lib/python3.8/site-packages/stix2/base.py", line 110, in _check_property
    six.raise_from(
  File "<string>", line 3, in raise_from
stix2.exceptions.InvalidValueError: Invalid value for Note 'object_refs': The type-specifying prefix 'x-mitre-tactic' for this property is not valid
```

**Comment by clenk** [comment]
Hi @isaisabel, this is definitely a bug and we will work on a fix. Thanks.

**Comment by clenk** [comment]
This should be fixed by #503, as long as `allow_custom=True` is used when initiating the MemoryStore: `src = MemoryStore(allow_custom=True)`.
